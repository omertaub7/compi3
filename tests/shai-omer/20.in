// different types passed into the function
byte byteFunc(byte x, byte y, int z, bool w) {
	if (w) {
		if (x - y < z) {
			return y - x;
		}
		else {
			return x - y;
		}
	}
	else {
		return x * y;
	}
}

void main() {
	byte x = byteFunc(10b, 5b, 4, true); // 5b
	printi(x); // 5
	byte y = byteFunc(10b, 7b, 4b, true); // 253b
	printi(y); // 253
	byte z = byteFunc(100b, 2b, 300 + 2b, (10 == 11));
	printi(z); // 200
}