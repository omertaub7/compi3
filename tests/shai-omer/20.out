5
253
200
